### PR TITLE
Add conversion functions for more `Legacy` types

### PR DESCRIPTION
**Pull request by jasagredo**
Part of #182

### CODE REVIEW

File: ouroboros-consensus-cardano-legacy-block/src/DBSync.hs [hunk 1]
```diff
@@ -239,3 +265,101 @@ instance Convert x (LegacyBlock x) where
...
+  . MismatchEraInfo
+  . unsafeCoerce
+    -- This should not be problematic because the LedgerEraInfo and
+    -- SingleEraInfo are just wrappers over a Text, so everything should be
+    -- safely coerced, but SOP complains about not being able to resolve the Top
+    -- constraint for every type.
+  . getMismatchEraInfo
+  $ we
+
+  -- where
```
Comment (dnadales) [inline ouroboros-consensus-cardano-legacy-block/src/DBSync.hs:330]: :zombie:

**Review by dnadales — APPROVED** [review]

**Comment by jorisdral** [comment]
I will be making some edits to this PR to move the conversion functions into a dedicated module, and to add a few more conversion functions

**Comment by jorisdral** [comment]
Done. Once this PR is re-approved, I will merge it even though CI isn't passing: we disabled the mempool tracers temporarily, which makes some mempool tests fail.

File: ouroboros-consensus-cardano-legacy-block/src/Legacy/Convert.hs [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+{-# LANGUAGE FlexibleInstances        #-}
+{-# LANGUAGE MultiParamTypeClasses    #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE StandaloneKindSignatures #-}
+{-# LANGUAGE TypeApplications         #-}
+{-# LANGUAGE TypeFamilies             #-}
+{-# LANGUAGE TypeOperators            #-}
+{-# LANGUAGE UndecidableInstances     #-}
+
+-- TODO: remove
```
Comment (dnadales) [inline ouroboros-consensus-cardano-legacy-block/src/Legacy/Convert.hs]: :zombie:

**Review by dnadales — APPROVED** [review]

**Pull request merged**